alf/networks/disjoint_policy_value_network.py

```diff
@@ -0,0 +1,263 @@
+# Copyright (c) 2021 Horizon Robotics and ALF Contributors. All Rights Reserved.
```

Comment:
I wonder do we need such a big class to hold several networks? It seems to me that these can be included in the algorithm.

Reply:
I think there are clear benefits about having it in a separate `.py` file:

1. It is cleaner this way as all network related logic are grouped together (so does all the documentation, the network class looks big but it is half documentation half implementation).
2. On the other side, if we put so many lines into the algorithm, it will make the algorithm much bigger and therefore much less readable. 
3. We can put unit test in a file that is specifically tied to this class.
4. We can hide implementations in this file so that people only cares about the algorithm don't have to read them.
5. We can hide the private/non-public methods/functions in this file so that they do not appear in the algorithm implementation, which can harm readability too.

I might have missed but did not see significant cost of having a separate `.py` file (or, a separate library).

Reply:
> I think there are clear benefits about having it in a separate `.py` file:
> 
> 1. It is cleaner this way as all network related logic are grouped together (so does all the documentation, the network class looks big but it is half documentation half implementation).
> 2. On the other side, if we put so many lines into the algorithm, it will make the algorithm much bigger and therefore much less readable.
> 3. We can put unit test in a file that is specifically tied to this class.
> 4. We can hide implementations in this file so that people only cares about the algorithm don't have to read them.
> 5. We can hide the private/non-public methods/functions in this file so that they do not appear in the algorithm implementation, which can harm readability too.
> 
> I might have missed but did not see significant cost of having a separate `.py` file (or, a separate library).

I think my point is that this network is a highly specialized one that probably won't be used by any other algorithm in the future. Other networks in alf.networks are very general. 

The details in this class might be critical for understanding PPG. If someone wants to understand PPG, he/she also needs to look at this file eventually.

Reply:
Agree that networks in this folder should be more generic.

Reply:
Right this network is high specialized, thought I still think it should be in its own `.py` file, I do agree it might be too specialized for `network`.

I have two proposals here:

1. put it under `networks/ppg/`
2. put it under `algorithms/ppg`

I understand that currently `algorithms/` is a flat directory, but having a secondary directory structure there can help grouping related files there. In this case we can have the network, `ppg_aux_phase_loss.py` and `ppg_algorithm.py`.

What do you think?